contracts/JBGenericBuybackDelegate.sol

```diff
@@ -165,229 +123,223 @@ contract JBGenericBuybackDelegate is
+    {
+        return (_data.reclaimAmount.value, _data.memo, delegateAllocations);
+    }
+
+    //*********************************************************************//
+    // -------------------------- public views --------------------------- //
+    //*********************************************************************//
+
+    function supportsInterface(bytes4 _interfaceId) public view override(ERC165, IERC165) returns (bool) {
+        return _interfaceId == type(IJBFundingCycleDataSource3_1_1).interfaceId
+            || _interfaceId == type(IJBPayDelegate3_1_1).interfaceId
+            || _interfaceId == type(IJBGenericBuybackDelegate).interfaceId
+            || super.supportsInterface(_interfaceId);
     }
 
-    /////////////////////////////////////////////////////////////////////
-    //                       External functions                        //
-    /////////////////////////////////////////////////////////////////////
-
-    /**
-     * @notice Delegate to either swap to the beneficiary or mint to the beneficiary
-     *
-     * @dev    This delegate is called only if the quote for the swap is bigger than the lowest received when minting.
-     *         If the swap reverts (slippage, liquidity, etc), the delegate will then mint the same amount of token as
-     *         if the delegate was not used.
-     *         If the beneficiary requests non claimed token, the swap is not used (as it is, per definition, claimed token)
-     *
-     * @param _data the delegate data passed by the terminal
-     */
+    //*********************************************************************//
+    // ---------------------- external transactions ---------------------- //
+    //*********************************************************************//
+
+    /// @notice Delegate used to swap a provided amount to the beneficiary, using any leftover amount to mint.
+    /// @dev This delegate is called only if the quote for the swap is bigger than the quote when minting.
+    /// If the swap reverts (slippage, liquidity, etc), the delegate will then mint the same amount of token as if the delegate was not used.
+    /// @param _data The delegate data passed by the terminal.
     function didPay(JBDidPayData3_1_1 calldata _data) external payable override {
-        // Access control as minting is authorized to this delegate
+        // Make sure only a payment terminal belonging to the project can access this functionality.
         if (!DIRECTORY.isTerminalOf(_data.projectId, IJBPaymentTerminal(msg.sender))) {
             revert JuiceBuyback_Unauthorized();
         }
 
-        (uint256 _tokenCount, uint256 _swapMinAmountOut, IERC20 _projectToken) =
-            abi.decode(_data.dataSourceMetadata, (uint256, uint256, IERC20));
+        // Parse the metadata passed in from the data source.
+        (
+            bool _quoteExists,
+            uint256 _minimumSwapAmountOut,
+            uint256 _weight,
+            bool _projectTokenIs0
```

Comment:
tiny-nit: changing the order so the bits are next to each other saves 1 slot in memory expansion